### PR TITLE
Add "shadow" param to auth hosting

### PR DESCRIPTION
**Pull request by allenzhou101**
## Related Issues

✅ Fixes https://github.com/descope/etc/issues/8364

## Description

💬 A few sentences describing the overall goals of the pull request's commits.

## Screenshots

📺🔫 All of the UI influenced by this PR

## Must

- [ ] 📱 Responsiveness (mobile/XL resolutions)
- [ ] 🧪 Tests
- [ ] 📃 Documentation (if applicable)

### CODE REVIEW

**Comment by gaokevin1** [comment]
@asafshen can you please review this PR for native flows.

File: src/App.tsx [hunk 1]
```diff
@@ -94,7 +94,9 @@ const App = () => {
 		flowId === 'saml-config' ||
 		flowId === 'sso-config';
 
-	const containerClasses = clsx('descope-base-container', {
+	const shadow = urlParams.get('shadow') !== 'false';
+
+	const containerClasses = clsx(shadow && 'descope-base-container', {
```
Comment (nirgur) [inline src/App.tsx:99]: I suggest that we put the class inside the object (like the other classes) to keep consistency

**Review by nirgur — APPROVED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -94,7 +94,9 @@ const App = () => {
 		flowId === 'saml-config' ||
 		flowId === 'sso-config';
 
-	const containerClasses = clsx('descope-base-container', {
+	const shadow = urlParams.get('shadow') !== 'false';
```
Comment (asafshen) [inline src/App.tsx:97]: in addition to Nir's comment, I think we consider how to automate detect this and not asking developers to provide

Comment (allenzhou101) [reply src/App.tsx:97]: But it's not necessarily just mobile right, people generally may want to remove background shadow but it's personal preference. So no way to detect that.

**Comment by allenzhou101** [comment]
ok resolved, @nirgur @asafshen

**Review by nirgur — APPROVED** [review]

**Comment by allenzhou101** [comment]
@omercnet Could you help with permissions here? Seems there's issues with at least Vercel

**Comment by gaokevin1** [comment]
@nirgur can we merge it?

We've done that in the past I think, since it's not a breaking change.

**Comment by allenzhou101** [comment]
@omercnet i can't bc i dont have write permissions to the repo...

that's why i had to fork it

**Comment by allenzhou101** [comment]
Screenshot 2024-12-08 at 4 19 16 PM

**Comment by allenzhou101** [comment]
was trying to get this out quick for you.com...

anyways maybe the best is for someone with write permissions to just make a new branch in the repo, @gaokevin1 @gshriki @asafshen @nirgur

**Comment by dorsha** [comment]
@omercnet can you fix the permissions?
Also we need to think how to handle external contributors...

**Comment by omercnet** [comment]
Permissions fixed

**Comment by nirgur** [comment]
opened another [PR](https://github.com/descope/auth-hosting/pull/632) with the same change

**Pull request closed without merging**